File: PascalTriangle.java
import java.util.ArrayList;
import java.util.List;

class Solution {
    public List<List<Integer>> pascalTriangle(int N) {
        // This stores our pascal's triangle
        List<List<Integer>> res = new ArrayList<>();

        // Iterating over each row
        for (int i = 0; i < N; i++) {
            // Resizing the array
            res.add(new ArrayList<>());
            res.get(i).add(1);

            // Assign 1 to the first and last location of this array
            for (int j = 1; j < i; j++) {
                // Find the integer present in i-1th row and j-1th column
                // and add it with the integer present in i-1th row and jth column
                int val = res.get(i - 1).get(j - 1) + res.get(i - 1).get(j);
                res.get(i).add(val);
            }

            if (i > 0) {
                res.get(i).add(1);
            }
        }

        return res;
    }
}

File: PascalTriangle.py
from typing import List

class Solution:
    def pascalTriangle(self, N: int) -> List[List[int]]:
        # This stores our pascal's triangle
        res = []

        # Iterating over each row
        for i in range(N):
            # Resizing the array
            res.append([])
            res[i].append(1)

            # Assign 1 to the first and last location of this array
            for j in range(1, i):
                # Find the integer present in i-1th row and j-1th column
                # and add it with the integer present in i-1th row and jth column
                val = res[i-1][j-1] + res[i-1][j]
                res[i].append(val)

            if i > 0:
                res[i].append(1)

        return res
